src/utils/document-model.ts

```diff
@@ -44,7 +44,7 @@ export function useDocumentDispatch<State, A extends Action, LocalState>(
     initialState: Document<State, A, LocalState>,
     onError: OnErrorHandler = console.error,
 ): readonly [
-    Document<State, A, LocalState>,
+    Document<State, A, LocalState> | undefined,
```

Comment:
There are quite a few places that check for the existence of the returned `document`, but this type says that it is always defined. I added this | undefined to warn that the document might not exist.